src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -196,10 +196,9 @@ trait EntityComponent {
 
       for {
         preExistingEntityRecs <- lookupEntitiesByNames(workspaceContext.workspaceId, entities.map(_.toReference))
-        _ <- deleteEntityAttributes(preExistingEntityRecs)
         savingEntityRecs <- insertNewEntities(workspaceContext, entities, preExistingEntityRecs).map(_ ++ preExistingEntityRecs)
         referencedAndSavingEntityRecs <- lookupNotYetLoadedReferences(workspaceContext, entities, savingEntityRecs).map(_ ++ savingEntityRecs)
-        _ <- insertAttributes(entities, referencedAndSavingEntityRecs)
+        _ <- upsertAttributes(entities, (savingEntityRecs ++ preExistingEntityRecs).distinct, referencedAndSavingEntityRecs)
```

Comment:
i have seen it to be the case but i'll remove it since it doesn't hurt performance but it will mean that we hit our max packet size ~twice as fast in the worst case